### PR TITLE
Fix Android breadcrumbs on web

### LINKED ISSUE #979: Breadcrumbs stopped working on Chrome
Using Dart `2.17.6`, Sentry `6.9.0` on the web, breadcrumbs stopped working on Chrome and Firefox. They still work fine on Safari.

It seems to be affecting all Chrome versions.

I'm not sure when it stopped working, perhaps around Sentry `6.0.0`, but not sure.

Just going to go ahead and pre-empt the inevitable question: This has nothing to do with Flutter. It's a Dart web app, that's it.

### PR DESCRIPTION
**Pull request by mnordine**
Fixes https://github.com/getsentry/sentry-dart/issues/979

### CODE REVIEW

**Comment by mnordine** [comment]
Sorry, I mistakenly added a bad initial commit with an extra line.

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,10 @@
 
 ### Fixes
 
+- Fix breadcrumbs not being sent on Android web ([#979](https://github.com/getsentry/sentry-dart/pull/1378))
```
Comment (marandaneto) [inline CHANGELOG.md]:
```suggestion
- Fix breadcrumbs not being sent on Android web ([#1378](https://github.com/getsentry/sentry-dart/pull/1378))
```

File: dart/lib/src/sentry_client.dart [hunk 2]
```diff
@@ -108,7 +108,8 @@ class SentryClient {
       return _sentryId;
     }
 
-    if (_options.platformChecker.platform.isAndroid &&
+    if (!_options.platformChecker.isWeb &&
+        _options.platformChecker.platform.isAndroid &&
```
Comment (marandaneto) [inline dart/lib/src/sentry_client.dart]: Can you check if this PR https://github.com/getsentry/sentry-dart/pull/1366/ would fix the issue as well?

Comment (marandaneto) [reply dart/lib/src/sentry_client.dart]: The PR still checks `isAndroid` and `isWeb` only but I don't see how `isWeb` can be true but also `isAndroid` at the same time.
@denrase have you tested on Web?

Comment (mnordine) [reply dart/lib/src/sentry_client.dart]: > I don't see how isWeb can be true but also isAndroid at the same time.

It certainly was, although I don't know the details.

Comment (mnordine) [reply dart/lib/src/sentry_client.dart]: Will check the other PR when I get a chance...

Comment (marandaneto) [reply dart/lib/src/sentry_client.dart]: @mnordine were you running a Flutter web app on a Mobile Android browser by any chance?

Comment (mnordine) [reply dart/lib/src/sentry_client.dart]: Nope, nothing to do with Flutter.

Comment (marandaneto) [reply dart/lib/src/sentry_client.dart]: So were you running a pure Dart Web app? Which browser? which OS?

Comment (mnordine) [reply dart/lib/src/sentry_client.dart]: It's right in the issue: https://github.com/getsentry/sentry-dart/issues/979

Dart web app, affects all browsers on Android.

Comment (marandaneto) [reply dart/lib/src/sentry_client.dart]: So still makes sense, @mnordine can you merge the main branch into your branch and fix the merge conflict? that part you changed moved to a different method.
Also please add a test :)
Thank you!

Comment (mnordine) [reply dart/lib/src/sentry_client.dart]: > Can you check if this PR https://github.com/getsentry/sentry-dart/pull/1366 would fix the issue as well?

Looks like the bug still exists when run against `main`, after merging in https://github.com/getsentry/sentry-dart/pull/1366.

> can you merge the main branch into your branch and fix the merge conflict?

I don't think I'll have a chance today, but will get to it when I can.

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,10 @@
 
 ### Fixes
 
+- Fix breadcrumbs not being sent on Android web ([#1378](https://github.com/getsentry/sentry-dart/pull/1378))
+
+### Fixes
+
```
Comment (marandaneto) [inline CHANGELOG.md]:
```suggestion
- Fix breadcrumbs not being sent on Android web ([#1378](https://github.com/getsentry/sentry-dart/pull/1378))
```

The Fixes section already exists.

File: dart/lib/src/sentry_client.dart (hunk 2 above)
Comment (mnordine) [reply dart/lib/src/sentry_client.dart]: > Also please add a test :)

I added a test, but I was kind of taking a shot in the dark, not being familiar with your test suite, practices, etc...

**Review by marandaneto — APPROVED** [review]
Thanks @mnordine

**Pull request merged**